tests/src/OneLogin/Saml2/SettingsTest.php

```diff
@@ -422,33 +422,78 @@ public function testGetSPMetadata()
     * Case with x509certNew
     *
     * @covers OneLogin_Saml2_Settings::getSPMetadata
+    * @dataProvider testGetSPMetadataWithX509CertNewDataProvider
```

Comment:
Since there were starting to be lots of specific cases to check for, I moved the cases out to a `@dataProvider`